src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -1655,25 +1653,51 @@ private void CheckModelAndSyntaxNodeToSpeculate(CSharpSyntaxNode syntax)
 
             info.Free();
 
-            if ((options & LookupOptions.IncludeExtensionMethods) != 0)
+            if ((options & LookupOptions.IncludeExtensionMethodsAndMembers) != 0)
             {
-                var lookupResult = LookupResult.GetInstance();
+                Debug.Assert((options & LookupOptions.MustBeInstance) == 0);
 
-                options |= LookupOptions.AllMethodsOnArityZero;
-                options &= ~LookupOptions.MustBeInstance;
+                var lookupResult = LookupResult.GetInstance();
+                options |= LookupOptions.AllMethodsOnArityZero | LookupOptions.AllNamedTypesOnArityZero;
 
                 var discardedUseSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
-                binder.LookupExtensionMethods(lookupResult, name, 0, options, ref discardedUseSiteInfo);
+
+                foreach (var scope in new ExtensionScopes(binder))
+                {
+                    // PROTOTYPE(static) confirm that we want to exclude type parameters from extension member resolution or leave a comment
+                    if (container is TypeSymbol typeSymbol && !typeSymbol.IsTypeParameter())
+                    {
+                        // When looking up members of an extension type, we don't need to find its members as part of
+                        // an extension member lookup since we'll already have found them as part of member lookup on the extension type.
+                        TypeSymbol skipType = typeSymbol.IsExtension ? typeSymbol.OriginalDefinition : null;
+
+                        // We use a Lookup API (which uses a CheckViability check) instead of an AddLookup API (which uses a CanAddLookupSymbolInfo check),
+                        // but that is fine since we filter symbols that cannot be referenced by name below anyways.
+                        scope.Binder.LookupImplicitExtensionMembersInSingleBinder(
```

Comment:
>scope.Binder.LookupImplicitExtensionMembersInSingleBinder( [](http://example.com/codeflow?start=24&length=58)

It looks like we are adding to the same ```lookupResult``` for all scopes. How is this going to interact with most specific shadowing?